### PR TITLE
[modern] Add simple tests for GCC and clang (build simple project)

### PR DESCRIPTION
**Pull request by jgsogo**
- [ ] Need to test actual linked libraries

### CODE REVIEW

File: modern/tests/test_build/test_simple.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        container.exec(['cmake', '--build', '.'])
+
+        # Check it runs
+        out, err = container.exec(['./example-c'])
+        assert 'Current local time and date' in out, f"out: '{out}' err: '{err}'"
+        out, err = container.exec(['./example-cpp'])
+        assert 'Current date' in out, f"out: '{out}' err: '{err}'"
+
+        # Check linked libs
+        out, err = container.exec(['ldd', 'example-c'])
```
Comment (jgsogo) [inline modern/tests/test_build/test_simple.py:26]: Output here for gcc10:

```
        linux-vdso.so.1 =>  (0x00007ffd08548000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f408fd98000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f408f9ce000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f408ffb5000)

```

File: modern/tests/test_build/test_simple.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        # TODO: Test linked libraries here (gcc)
+        print(out)
+        print(err)
+
+        out, err = container.exec(['ldd', 'example-cpp'])
```
Comment (jgsogo) [inline modern/tests/test_build/test_simple.py:31]: Output here for gcc10:

```
        linux-vdso.so.1 =>  (0x00007fff2b8df000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa4e83c2000)
        libstdc++.so.6 => /usr/local/lib64/libstdc++.so.6 (0x00007fa4e7fbc000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa4e7cb3000)
        libgcc_s.so.1 => /usr/local/lib64/libgcc_s.so.1 (0x00007fa4e7a9c000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa4e76d2000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fa4e85df000)
```

File: modern/tests/test_build/test_simple.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        print(err)
+
+    # Check we can run these executables in vanilla image
+    vanilla_img = f"{expected.distro.name}:{expected.distro.version}"
+    with run_container(vanilla_img, tmpdirname=container._tmpfolder) as vanilla:
+        with vanilla.working_dir(build_directory):
+            out, err = vanilla.exec(['./example-c'])
+            assert 'Current local time and date' in out, f"out: '{out}' err: '{err}'"
+
+            out, err = vanilla.exec(['./example-cpp'])
```
Comment (jgsogo) [inline modern/tests/test_build/test_simple.py:43]: Error here:

```
./example-cpp: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by ./example-cpp)
```

**Comment by uilianries** [comment]
It seems like simple folder requires some extra permission:

Running on my local Linux machine:
```
$ git checkout tests/simple
fatal: cannot create directory at 'modern/tests/workingdir/simple': Permission denied
```

**Comment by uilianries** [comment]
> It seems like simple folder requires some extra permission:
> 
> Running on my local Linux machine:
> 
> ```
> $ git checkout tests/simple
> fatal: cannot create directory at 'modern/tests/workingdir/simple': Permission denied
> ```

Forget about, cloning directly from your Fork work.

File: modern/tests/test_build/test_simple.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        # C executable should run in all the containers
+        image_name = expected.image_name(compiler, Version(compiler_version))
+        with container.run_container(image_name) as image:
+            with image.working_dir(self._build_directory):
+                out, err = image.exec(['./example-c'])
+                assert 'Current local time and date' in out, f"out: '{out}' err: '{err}'"
+
+    @pytest.mark.service('xtest')
+    @pytest.mark.parametrize("compiler, compiler_version", [(key, str(v)) for key, value in get_compiler_versions().items() for v in value])
+    def test_cpp_compatible(self, container, expected, compiler, compiler_version):
```
Comment (jgsogo) [inline modern/tests/test_build/test_simple.py:63]: Need to review, probably it requires another PR

File: modern/tests/conftest.py [hunk 5]
```diff
@@ -1,8 +1,10 @@
-import os
-
 import pytest
 
-docker_compose_services = ['base', 'builder', 'deploy', 'jenkins']
+docker_compose_services = ['base',
+                           # 'builder',  This is just a helper image, do we need to test it?
```
Comment (SSE4) [inline modern/tests/conftest.py:4]: I don't think so

**Review by uilianries — APPROVED** [review]

**Pull request merged**